substratevm/src/com.oracle.svm.test/src/com/oracle/svm/test/jfr/JFRTest.java

```diff
@@ -40,12 +39,16 @@
 import com.oracle.svm.test.jfr.utils.JFRFileParser;
 import com.oracle.svm.test.jfr.utils.LocalJFR;
 
+import java.util.HashMap;
 import jdk.jfr.Recording;
+import jdk.jfr.consumer.RecordedEvent;
 import jdk.jfr.consumer.RecordingFile;
 
 /** Base class for JFR unit tests. */
 public abstract class JFRTest {
 
+    private HashMap<String, Boolean> eventMap = new HashMap<>();
```

Comment:
This field can be converted to a local in `checkEvents()` (`HashSet<String> seenEvents`) as there is not really any need to prefill this map. `checkEvents()` can then check if all the expected event types are in this set.

Reply:
Fixed